lib/view_component/translatable.rb

```diff
@@ -23,9 +23,13 @@ def i18n_scope
       def build_i18n_backend
         return if compiled?
 
-        self.i18n_backend = if (translation_files = sidecar_files(%w[yml yaml])).any?
-          # Returning nil cleans up if translations file has been removed since the last compilation
+        # We need to load the translations files from the ancestors so a component
+        # can inherit translations from its parent and is able to overwrite them.
+        vc_ancestors = ancestors.reverse.select { _1.is_a?(Class) && _1 < ViewComponent::Base }
+        translation_files = vc_ancestors.flat_map { _1.sidecar_files(%w[yml yaml]) }
```

Comment:
```suggestion
        translations = ancestors.each_with_object([]) do |ancestor, sidecar_files|
          sidecar_files += side_car_files(%w[yml yaml]) if ancestor.is_a?(Class) && ancestor < ViewComponent::Base
        end
```

Thoughts on optimizing this to be a single loop? I imagine this will be called pretty often and I think it's still readable, so likely worth it?

Reply:
Sure! I'll update 👍